src/Tokenomia/Vesting/PrivateSale.hs

```diff
@@ -0,0 +1,327 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE NoImplicitPrelude   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
+{-# LANGUAGE BlockArguments      #-}
+
+module Tokenomia.Vesting.PrivateSale (verifyPrivateSale) where
+
+import           Prelude                (IO, Semigroup (..), Show (..), String)
+import qualified PlutusTx
+import           PlutusTx.Prelude       hiding (Semigroup(..), unless)
```

Comment:
Maybe it makes more sense to import `Prelude hiding` a few things and only import very little from `PlutusTx.Prelude`?

Reply:
Agreed here, since we're mostly using blockfrost, lets use as much of standard prelude as possible